pkg/stores/proxy/cache/cache.go

```diff
@@ -0,0 +1,147 @@
+// Package cache implements the sized revision cache. The size revision cache maintains a cache of Unstructuredlists while
+// attempting to limit the aggregate values stored by the given memory size (in bytes) parameter.
+package cache
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/cache"
+)
+
+var (
+	NotFoundErr = errors.New("key was not found in cache")
+)
+
+// SizedRevisionCache maintains a cache of Unstructuredlists while attempting to limit the aggregate values stored by
+// the given memory size (in bytes) parameter.
+type SizedRevisionCache struct {
+	listRevisionCache *cache.LRUExpireCache
+	cacheLock         sync.Mutex
+	size              int
+	sizeLimit         int
+}
+
+// Cacher is a cache that stores and retrieves UnstructuredLists.
+type Cacher interface {
+	Get(key cacheKey) (*unstructured.UnstructuredList, error)
+	Add(key cacheKey, list *unstructured.UnstructuredList) error
+}
+
+type cacheKey struct {
+	resourcePath string
+	revision     string
+	namespace    string
+	cont         string
+}
```

Comment:
I think other query parameters from https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ListOptions need to be included here.

For example, if `limit` is set to 10 on one request, then a list with 10 items will be cached. If the next request sets limit to `100`, the cache key would still match the last and only 10 items would be returned. Similarly for `fieldSelector` or `labelSelector`.

This is already a bug in the current cache implementation :(

Reply:
Went ahead and updated this. Kept it in a separate commit for now. I can reset and squash the parts into their respective commit if we're happy with it.